src/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeReader.cs

```diff
@@ -277,22 +292,14 @@ public override bool TryRead(out ReadResult result)
         {
             ThrowIfCompleted();
 
-            return TryReadInternal(InternalTokenSource, out result);
+            return TryReadInternal(out result);
         }
 
-        private bool TryReadInternal(CancellationTokenSource source, out ReadResult result)
+        private bool TryReadInternal(out ReadResult result)
         {
-            bool isCancellationRequested = source.IsCancellationRequested;
-            if (isCancellationRequested || _bufferedBytes > 0 && (!_examinedEverything || _isStreamCompleted))
+            if (_awaitable.IsCompleted || (_bufferedBytes > 0 && (_awaitable.IsCompleted || _isStreamCompleted)))
```

Comment:
You know _awaitable is not completed in the second half of this condition:

```suggestion
            if (_awaitable.IsCompleted || (_bufferedBytes > 0 && _isStreamCompleted))
```

Reply:
Also, should TryRead check _streamReadTask somewhere and throw if it's not completed?